### ISSUE TITLE
How can I make this work with a custom top layer?

### ISSUE THREAD

**Issue by Dsanzv** [keisen/tf-keras-vis#10]
Hello. I'm new to ML and newer still to this library. I want to use the gradcam on a vgg16 architecture with custom top layers that I created. If my model in this case consists of two separate models, what would I have to change, say, in the example case to make this work?

**Comment by keisen** [comment]
> If my model in this case consists of two separate models

Could you please explain that to me a little more detail ?
Or could you submit code snippet to me to reproduce your models ?

**Comment by Dsanzv** [comment]
> 
> 
> > If my model in this case consists of two separate models
> 
> Could you please explain that to me a little more detail ?
> Or could you submit code snippet to me to reproduce your models ?



> 
> 
> > If my model in this case consists of two separate models
> 
> Could you please explain that to me a little more detail ?
> Or could you submit code snippet to me to reproduce your models ?

Hey, thanks for the quick response. Here's the portion of my code.
`Loading vgc16 model
vgg16 = applications.VGG16(include_top=False, weights='imagenet')


model = Sequential()  
model.add(Flatten(input_shape=train_data.shape[1:]))  
model.add(Dense(100))
model.add(LeakyReLU(alpha=0.03))
model.add(Dropout(0.5))  
model.add(Dense(50))
model.add(LeakyReLU(alpha=0.03))
model.add(Dropout(0.3)) 
model.add(Dense(num_classes, activation='softmax')) `

So even though the two parts work, they're two separate models and I'm still not very good at dealing with them. What would I have to do differently to make these work?

**Comment by keisen** [comment]
That's, do you want to do such as follow ?

```
base_model = tf.keras.applications.VGG16(include_top=False)
x = base_model.output
x = Dense(100)(x)
x = LeakyReLU(alpha=0.03)(x)
x = Dropout(0.5)(x)
x = Dense(50)(x)
x = LeakyReLU(alpha=0.03)(x)
x = Dropout(0.3)(x)
x = Dense(num_classes, activation='softmax')(x)

model = tf.keras.Model(base_model.input, x)
```

**Comment by Dsanzv** [comment]
Again, thanks for the time taken to respond to me. No, I think I can't do that because I'm developing a model to train under very limited computational capabilities and I'm dividing the training into the feature extraction from the datasets and then the upper layers classification. It's really good for quick adjustments and such, but I haven't been able to make it work well with the heatmap visualisation methods I've seen so far

**Comment by keisen** [comment]
> I'm developing a model to train under very limited computational capabilities and I'm dividing the training into the feature extraction from the datasets and then the upper layers classification.

Why Couldn't you combine the models when inferencing ?
I believe that, if you could combine the models as one model, you can visualize heatmap.

Did I misunderstand your question?

**Comment by Dsanzv** [comment]
> Why Couldn't you combine the models when inferencing ?
> I believe that, if you could combine the models as one model, you can visualize heatmap.
> 
> Did I misunderstand your question?

No, I think you got it. It's just that I don't know how. I can use both sequentially but I don't how to combine them into a single model in the inferencing stage. Sorry for the asinine questions, I know this has a really simple solution but I can't quite grasp it.

**Comment by keisen** [comment]
Sorry for the confusion!, I did misunderstand about your situation.

You can combine the models using `tf.keras.Sequential` that allow to add the model as layer,
but can NOT use Gradcam to the model basically. Because tf.keras can not return the gradients of cascaded model.

The only way is that you define a new unified model and then copy weights of trained models to new one. But I don't recommend it basically, because it's too complicated.

**Comment by Dsanzv** [comment]
I see. So the best solution possible if I want to use your toolkit would be to get a better machine for the training process and put all of the layers together, right?

**Comment by keisen** [comment]
Yes, absolutely.
